src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ProvideStreamMemoryAsyncOverridesTests.cs

```diff
@@ -0,0 +1,1476 @@
+
+            var test = new VerifyVB.Test
+            {
+                TestCode = code,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                ExpectedDiagnostics =
+                {
+                    VerifyVB.Diagnostic(Rule)
+                        .WithLocation(0)
+                        .WithArguments("River", VBDisplayWriteAsyncArray, VBDisplayWriteAsyncMemory),
+                    VerifyVB.Diagnostic(Rule)
+                        .WithLocation(0)
+                        .WithArguments("River", VBDisplayReadAsyncArray, VBDisplayReadAsyncMemory)
+
+                }
+            };
+            return test.RunAsync();
+        }
+
+        [Theory]
+        [InlineData(CSReadAsyncArray, CSDisplayReadAsyncArray, CSDisplayReadAsyncMemory)]
+        [InlineData(CSWriteAsyncArray, CSDisplayWriteAsyncArray, CSDisplayWriteAsyncMemory)]
+        public Task SingleArrayOverride_MultiplePartialsInSameFile_ReportsAllLocations_CS(string arrayMethod, string displayArrayMethod, string displayMemoryMethod)
+        {
+            string code = $@"
+{CSUsings}
+namespace Testopolis
+{{
+    public partial class {{|#0:River|}} : Stream
+    {{
+        {CSAbstractMembers}
+    }}
+}}
+
+namespace Testopolis
+{{
+    partial class {{|#1:River|}}
+    {{
+        {arrayMethod}
+    }}
+}}";
+
+            var test = new VerifyCS.Test
+            {
+                TestCode = code,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                ExpectedDiagnostics =
+                {
+                    VerifyCS.Diagnostic(Rule)
+                        .WithLocation(0)
+                        .WithLocation(1)
+                        .WithArguments("River", displayArrayMethod, displayMemoryMethod),
+                }
+            };
+            return test.RunAsync();
+        }
+
+        [Theory]
+        [InlineData(VBReadAsyncArray, VBDisplayReadAsyncArray, VBDisplayReadAsyncMemory)]
+        [InlineData(VBWriteAsyncArray, VBDisplayWriteAsyncArray, VBDisplayWriteAsyncMemory)]
+        public Task SingleArrayOverride_MultiplePartialsInSameFile_ReportsAllLocations_VB(string arrayMethod, string displayArrayMethod, string displayMemoryMethod)
+        {
+            string code = $@"
+{VBUsings}
+Namespace Testopolis
+    Partial Public Class {{|#0:River|}} : Inherits Stream
+        {VBAbstractMembers}
+        {arrayMethod}
+    End Class
+End Namespace
+
+Namespace Testopolis
+    Partial Class {{|#1:River|}}
+    End Class
+End Namespace";
+
+            var test = new VerifyVB.Test
+            {
+                TestCode = code,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                ExpectedDiagnostics =
+                {
+                    VerifyVB.Diagnostic(Rule)
+                        .WithLocation(0)
+                        .WithLocation(1)
+                        .WithArguments("River", displayArrayMethod, displayMemoryMethod)
```

Comment:
This is very confusing. Is this supposed to be two diagnostics?
I believe that only one diagnostic is passed to the test.

Reply:
The location that is underlined when this diagnostic is reported is the name of the class in the class declaration. If the class is a partial class, all partial declarations need to be underlined. That is what this test is checking: that when you have multiple partial declarations in the same file, all partial declarations will be underlined for a single violation.

Reply:
@NewellClark If I understand correctly, you're passing only one diagnostic. When you chain `WithLocation` calls, only the latest one takes effect.

I believe if you removed the chain and kept the latest WithLocation call only, the test will still pass. Could you confirm that locally on your machine?

Reply:
Just tested it. The test fails if we remove any of the `WithLocation` calls, and the error message reports the diagnostics at the removed locations as the unexpected diagnostics.

Reply:
@NewellClark Good to know! Thanks.

Reply:
@Youssef1313 by the way, would it be possible to have this pull request linked to [issue 33789](https://github.com/dotnet/runtime/issues/33789)?

Reply:
@NewellClark The issue won't be auto-closed when the PR is merged. But a Microsoft employee will close it when he merges your PR.